src/google/protobuf/compiler/cpp/cpp_options.h

```diff
@@ -42,14 +42,21 @@ class AccessInfoMap;
 
 namespace cpp {
 
+enum class EnforceOptimizeMode {
+  kNoEnforcement, // Use the runtime specified by the file specific options.
+  kSpeed, // This is the full runtime.
+  kCodeSize,
```

Comment:
Could you actually leave out the code-size mode for now? That mode seems to be only rarely useful, so I think we should avoid making it more visible. We can always add a flag for it in the future if someone has a good use case for it.

Reply:
Done.